src/server/main_service.cc

```diff
@@ -68,6 +68,104 @@ std::optional<VarzFunction> engine_varz;
+  // This code is based on Redis impl for it at sdscatrepr@sds.c
+  std::string result = absl::StrCat("\"");
+
+  for (auto c : str) {
+    switch (c) {
+      case '\\':
+        absl::StrAppend(&result, "\\\\");
+        break;
+      case '"':
+        absl::StrAppend(&result, "\\\"");
+        break;
+      case '\n':
+        absl::StrAppend(&result, "\\n");
+        break;
+      case '\r':
+        absl::StrAppend(&result, "\\r");
+        break;
+      case '\t':
+        absl::StrAppend(&result, "\\t");
+        break;
+      case '\a':
+        absl::StrAppend(&result, "\\a");
+        break;
+      case '\b':
+        absl::StrAppend(&result, "\\b");
+        break;
+      default:
+        if (isprint(c)) {
+          result += c;
+        } else {
+          absl::StrAppend(&result, "\\x", absl::Hex((unsigned char)c, absl::kZeroPad2));
+        }
+        break;
+    }
+  }
+  absl::StrAppend(&result, "\"");
+  return result;
+}
+
+std::string MakeMonitorMessage(const ConnectionState& conn_state,
+                               const facade::Connection* connection, CmdArgList args) {
+  std::string message = CreateMonitorTimestamp();
+
+  if (conn_state.script_info.has_value()) {
+    absl::StrAppend(&message, "lua] ");
+  } else {
+    absl::StrAppend(&message, connection->RemoteEndpointStr());
+  }
+  if (args.empty()) {
+    absl::StrAppend(&message, "error - empty cmd list!");
+  } else if (auto cmd_name = std::string_view(args[0].data(), args[0].size());
+             cmd_name == "AUTH") {  // we cannot just send auth details in this case
+    absl::StrAppend(&message, "\"", cmd_name, "\"");
+  } else {
+    message = std::accumulate(args.begin(), args.end(), message, [](auto str, const auto& cmd) {
+      absl::StrAppend(&str, " ", CmdEntryToMonitorFormat(std::string_view(cmd.data(), cmd.size())));
+      return str;
+    });
+  }
+  return message;
+}
+
+void DispatchMonitorIfNeeded(bool admin_cmd, ConnectionContext* connection, CmdArgList args) {
+  // We are not sending any admin command in the monitor, and we do not want to
+  // do any processing if we don't have any waiting connections with monitor
+  // enabled on them - see https://redis.io/commands/monitor/
+  const auto& my_monitors = ServerState::tlocal()->Monitors();
+  if (!(my_monitors.Empty() || admin_cmd)) {
+    //  We have connections waiting to get the info on the last command, send it to them
+    auto monitor_msg = MakeMonitorMessage(connection->conn_state, connection->owner(), args);
+    // Note that this is accepted by value for lifetime reasons
+    // we want to have our own copy since we are assuming that
+    // 1. there will be not to many connections that we in monitor state
+    // 2. we need to have for each of them each own copy for thread safe reasons
+    VLOG(1) << "sending command '" << monitor_msg << "' to the clients that registered on it";
+    shard_set->pool()->DispatchBrief([monitor_msg](unsigned idx, util::ProactorBase*) {
```

Comment:
`[monitor_msg = std::move(monitor_msg)]`

Reply:
move it is